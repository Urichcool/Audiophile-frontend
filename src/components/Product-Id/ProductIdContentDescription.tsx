import { FC } from "react";
import Picture from "../Reusable-Components/Picture";
import { priceWithCommas } from "../../utils/priceWithCommas";
import AddToCartForm from "./AddToCartForm";

interface IProductIdContentDescriptionProps {
  pictureSrc?: { desktop: string; tablet: string; mobile: string };
  name?: string;
  isNew?: boolean;
  description?: string;
  price?: number;
  productId?: string;
  previewImage?: string;
}

const ProductIdContentDescription: FC<IProductIdContentDescriptionProps> = ({
  pictureSrc,
  name,
  isNew,
  description,
  price,
  productId,
  previewImage,
}) => {
  return (
    <div className="product-id-page-picture-and-description-container">
      <div className="product-id-page-picture-container">
        <Picture
          imageSrc={{
            desktop: pictureSrc?.desktop,
            tablet: pictureSrc?.tablet,
            mobile: pictureSrc?.mobile,
          }}
          sizes={{
            desktopWidth: "400",
            desktopHeight: "400",
            tabletWidth: "281",
            tabletHeight: "450",
            mobileWidth: "300",
            mobileHeight: "300",
          }}
          name={name}
          className=""
        />
      </div>
      <div className="product-id-page-description-container">
        {isNew && (
          <p className="Overline-manrope-regular product-id-page-new-product-text">
            NEW PRODUCT
          </p>
        )}
        <h2
          className="H2-manrope-bold product-id-page-product-title"
          data-testid="product-name"
        >
          {name}
        </h2>
        <p className="Body-manrope-medium product-id-page-product-description">
          {description}
        </p>
        <p className="product-id-page-product-price">
          {priceWithCommas(price)}
        </p>
        <AddToCartForm
          productId={productId}
          name={name}
          price={price}
          picture={previewImage}
        />
      </div>
    </div>
  );
};

export default ProductIdContentDescription;
