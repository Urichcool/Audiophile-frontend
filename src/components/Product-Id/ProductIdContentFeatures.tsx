import { FC } from "react";

interface IProductIdContentFeaturesProps {
  features?: string;
  includes?: { quantity: number; item: string; _id: string }[];
}

 const ProductIdContentFeatures: FC<IProductIdContentFeaturesProps> = ({
  features,
  includes,
}) => {
  return (
    <div className="product-id-page-features-and-includes-container">
      <div className="product-id-page-features-container">
        <h3 className="H3-manrope-bold product-id-page-features-title">
          Features
        </h3>
        <p className="product-id-page-features-text Body-manrope-medium">
          {features}
        </p>
      </div>
      <div className="product-id-page-includes-container">
        <h3 className="H3-manrope-bold product-id-page-includes-title">
          in the box
        </h3>
        <ul className="product-id-page-includes-list">
          {includes?.map(({ quantity, item, _id }) => (
            <li className="product-id-page-includes-list-item" key={_id}>
              <p className="product-id-page-includes-list-item-quantity-text">{`${quantity}x`}</p>
              <p className="product-id-page-includes-list-item-text Body-manrope-medium">
                {item}
              </p>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default ProductIdContentFeatures;
