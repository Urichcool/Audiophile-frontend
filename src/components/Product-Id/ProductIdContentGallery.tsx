import { FC } from "react";
import Picture from "../Reusable-Components/Picture";

interface IProductIdContentGalleryProps {
  gallery?: {
    first: { mobile: string; tablet: string; desktop: string };
    second: { mobile: string; tablet: string; desktop: string };
    third: { mobile: string; tablet: string; desktop: string };
  };
  name?: string;
}

const ProductIdContentGallery: FC<IProductIdContentGalleryProps> = ({
  gallery,
  name,
}) => {
  return (
    <div className="product-id-page-gallery">
      <div>
        <Picture
          imageSrc={{
            desktop: gallery?.first.desktop,
            tablet: gallery?.first.tablet,
            mobile: gallery?.first.mobile,
          }}
          sizes={{
            desktopWidth: "445",
            desktopHeight: "282",
            tabletWidth: "277",
            tabletHeight: "174",
            mobileWidth: "327",
            mobileHeight: "174",
          }}
          name={name}
          imgClassName="product-id-page-gallery-picture-1"
        />
        <Picture
          imageSrc={{
            desktop: gallery?.second.desktop,
            tablet: gallery?.second.tablet,
            mobile: gallery?.third.mobile,
          }}
          sizes={{
            desktopWidth: "445",
            desktopHeight: "282",
            tabletWidth: "277",
            tabletHeight: "174",
            mobileWidth: "327",
            mobileHeight: "174",
          }}
          name={name}
          imgClassName="product-id-page-gallery-picture-2"
        />
      </div>
      <Picture
        imageSrc={{
          desktop: gallery?.third.desktop,
          tablet: gallery?.third.tablet,
          mobile: gallery?.third.mobile,
        }}
        sizes={{
          desktopWidth: "635",
          desktopHeight: "586",
          tabletWidth: "395",
          tabletHeight: "370",
          mobileWidth: "327",
          mobileHeight: "370",
        }}
        name={name}
        imgClassName="product-id-page-gallery-picture-3"
      />
    </div>
  );
};

export default ProductIdContentGallery;
