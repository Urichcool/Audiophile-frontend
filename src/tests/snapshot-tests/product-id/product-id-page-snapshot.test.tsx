import ProductIdContentDescription from "../../../components/Product-Id/ProductIdContentDescription";
import ProductIdContentFeatures from "../../../components/Product-Id/ProductIdContentFeatures";
import ProductIdContentGallery from "../../../components/Product-Id/ProductIdContentGallery";
import { renderWithReduxAndRouter } from "../../helpers/RenderWithReduxAndRouter";
import {
  productIdContentDescriptionTestProps,
  productIdContentFeaturesTestProps,
  productIdContentGalleryTestProps,
} from "./product-id-page-test-props";
import { screen } from "@testing-library/react";

describe("product-id-page-snapshot-tests", () => {
  test("product-id-page-description-should-match-snapshot", () => {
    const {
      pictureSrc,
      name,
      isNew,
      description,
      price,
      productId,
      previewImage,
      category
    }: {
      pictureSrc: { desktop: string; tablet: string; mobile: string };
      name: string;
      isNew: boolean;
      description: string;
      price: number;
      productId: string;
        previewImage: string;
        category: string;
    } = productIdContentDescriptionTestProps;
    renderWithReduxAndRouter(
      <ProductIdContentDescription
        pictureSrc={pictureSrc}
        name={name}
        isNew={isNew}
        description={description}
        price={price}
        productId={productId}
        previewImage={previewImage}
        category={category}
      />
    );
    const productIdDescription: HTMLDivElement = screen.getByTestId(
      "product-id-page-picture-and-description-container"
    );
    expect(productIdDescription).toMatchSnapshot();
  });
  test("product-id-page-features-should-match-snapshot", () => {
    const { features, includes } = productIdContentFeaturesTestProps;
    renderWithReduxAndRouter(
      <ProductIdContentFeatures features={features} includes={includes} />
    );
    const productIdFeatures: HTMLDivElement = screen.getByTestId(
      "product-id-page-features-and-includes"
    );
    expect(productIdFeatures).toMatchSnapshot();
  });
  test("product-id-page-gallery-should-match-snapshot", () => {
    const { gallery, name } = productIdContentGalleryTestProps;
    renderWithReduxAndRouter(
      <ProductIdContentGallery gallery={gallery} name={name} />
    );
    const productIdGallery: HTMLDivElement = screen.getByTestId(
      "product-id-page-gallery"
    );
    expect(productIdGallery).toMatchSnapshot();
  });
});
