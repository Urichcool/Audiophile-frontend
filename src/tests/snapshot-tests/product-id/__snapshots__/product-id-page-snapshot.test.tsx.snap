// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-id-page-snapshot-tests product-id-page-description-should-match-snapshot 1`] = `
<div
  class="product-id-page-picture-and-description-container"
  data-testid="product-id-page-picture-and-description-container"
>
  <div
    class="product-id-page-picture-container"
  >
    <picture
      class=""
      data-testid="picture-component"
    >
      <source
        height="400"
        media="(min-width: 1440px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443139/audiophile/goods/product-xx99-mark-two-headphones/desktop/image-product-removebg-desktop-removebg-preview_myfyr1.png"
        width="400"
      />
      <source
        height="450"
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443187/audiophile/goods/product-xx99-mark-two-headphones/tablet/image-product-removebg-tablet-removebg-preview_wbyhs5.png"
        width="281"
      />
      <img
        alt="XX99 Mark II Headphones"
        height="300"
        src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443154/audiophile/goods/product-xx99-mark-two-headphones/mobile/image-product-removebg-mobile-removebg-preview_e3ysuj.png"
        width="300"
      />
    </picture>
  </div>
  <div
    class="product-id-page-description-container"
  >
    <h2
      class="H2-manrope-bold product-id-page-product-title"
      data-testid="product-name"
    >
      XX99 Mark II Headphones
    </h2>
    <p
      class="Body-manrope-medium product-id-page-product-description"
    >
      XX99 Mark II Headphones
    </p>
    <p
      class="product-id-page-product-price"
    >
      $ 499
    </p>
    <form
      action="#"
      class="add-to-cart-form"
    >
      <button
        class="add-to-cart-input-minus-button"
        data-testid="add-to-cart-form-decrease-button"
        type="button"
      >
        -
      </button>
      <input
        class="add-to-cart-input"
        data-testid="add-to-cart-form-quantity"
        id="quantity"
        name="quantity"
        readonly=""
        type="number"
        value="1"
      />
      <button
        class="add-to-cart-input-plus-button"
        data-testid="add-to-cart-form-increase-button"
        type="button"
      >
        +
      </button>
      <button
        class="add-to-cart-button"
        data-testid="add-to-cart-button"
        type="submit"
      >
        Add to cart
      </button>
    </form>
  </div>
</div>
`;

exports[`product-id-page-snapshot-tests product-id-page-features-should-match-snapshot 1`] = `
<div
  class="product-id-page-features-and-includes-container"
  data-testid="product-id-page-features-and-includes"
>
  <div
    class="product-id-page-features-container"
  >
    <h3
      class="H3-manrope-bold product-id-page-features-title"
    >
      Features
    </h3>
    <p
      class="product-id-page-features-text Body-manrope-medium"
    >
      Experience unrivalled stereo sound thanks to innovative acoustic technâ€¦
    </p>
  </div>
  <div
    class="product-id-page-includes-container"
  >
    <h3
      class="H3-manrope-bold product-id-page-includes-title"
    >
      in the box
    </h3>
    <ul
      class="product-id-page-includes-list"
    >
      <li
        class="product-id-page-includes-list-item"
      >
        <p
          class="product-id-page-includes-list-item-quantity-text"
        >
          2x
        </p>
        <p
          class="product-id-page-includes-list-item-text Body-manrope-medium"
        >
          Earphone unit
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`product-id-page-snapshot-tests product-id-page-gallery-should-match-snapshot 1`] = `
<div
  class="product-id-page-gallery"
  data-testid="product-id-page-gallery"
>
  <div>
    <picture
      data-testid="picture-component"
    >
      <source
        height="282"
        media="(min-width: 1440px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/desktop/oycf1je8ndux38npvk6a.jpg"
        width="445"
      />
      <source
        height="174"
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496516/audiophile/goods/product-yx1-earphones/tablet/cm2w05667a5uuvfs8aji.jpg"
        width="277"
      />
      <img
        alt="XX59 Headphones"
        class="product-id-page-gallery-picture-1"
        height="174"
        src="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/mobile/vzmpaa7dkwji3u3nk7co.jpg"
        width="327"
      />
    </picture>
    <picture
      data-testid="picture-component"
    >
      <source
        height="282"
        media="(min-width: 1440px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/desktop/dcifyu9zydbtrklw8gcr.jpg"
        width="445"
      />
      <source
        height="174"
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496516/audiophile/goods/product-yx1-earphones/tablet/pwcuqvrhxa779epydciq.jpg"
        width="277"
      />
      <img
        alt="XX59 Headphones"
        class="product-id-page-gallery-picture-2"
        height="174"
        src="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/mobile/rz9xwt6jbogh27ezrpt6.jpg"
        width="327"
      />
    </picture>
  </div>
  <picture
    data-testid="picture-component"
  >
    <source
      height="586"
      media="(min-width: 1440px)"
      srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/desktop/ndaegpi3cjyf8wfzpvj5.jpg"
      width="635"
    />
    <source
      height="370"
      media="(min-width: 768px)"
      srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496516/audiophile/goods/product-yx1-earphones/tablet/x3sscr6aajfe3zg5d2zd.jpg"
      width="395"
    />
    <img
      alt="XX59 Headphones"
      class="product-id-page-gallery-picture-3"
      height="370"
      src="https://res.cloudinary.com/dir4brsjf/image/upload/v1697496515/audiophile/goods/product-yx1-earphones/mobile/rz9xwt6jbogh27ezrpt6.jpg"
      width="327"
    />
  </picture>
</div>
`;
